pkg/operator/controller/ingress/nodeport_service.go

```diff
@@ -47,15 +47,17 @@ func (r *reconciler) ensureNodePortService(ic *operatorv1.IngressController, dep
 			return false, nil, fmt.Errorf("failed to create NodePort service: %v", err)
 		}
 		log.Info("created NodePort service", "service", desired)
+		return r.currentNodePortService(ic)
 	case wantService && haveService:
 		if updated, err := r.updateNodePortService(current, desired); err != nil {
 			return true, current, fmt.Errorf("failed to update NodePort service: %v", err)
 		} else if updated {
 			log.Info("updated NodePort service", "service", desired)
+			return r.currentNodePortService(ic)
 		}
 	}
 
-	return r.currentNodePortService(ic)
+	return true, current, nil
```

Comment:
We should still `return r.currentNodePortService(ic)` or possibly `return false, nil, nil` after line 44 under the `!wantService && haveService` case where we have just deleted the service.

Reply:
whoops, totally missed that. Thanks!